spec/components/search_spec.rb

```diff
@@ -368,6 +367,98 @@ def create_pm(users:, group: nil)
     end
   end
 
+  context 'posts' do
+    let(:post) { Fabricate(:post) }
+    let(:topic) { post.topic }
+
+    let!(:reply) do
+      Fabricate(:post_with_long_raw_content,
+        topic: topic,
+        user: topic.user,
+      ).tap { |post| post.update!(raw: "#{post.raw} elephant") }
+    end
+
+    let(:expected_blurb) do
+      "...quire content longer than the typical test post raw content. It really is some long content, folks. elephant"
+    end
+
+    it 'returns the post' do
+      result = Search.execute('elephant',
+        type_filter: 'topic',
+        include_blurbs: true
+      )
+
+      expect(result.posts).to contain_exactly(reply)
+      expect(result.blurb(reply)).to eq(expected_blurb)
+    end
+
+    it 'returns the right post and blurb for searches with phrase' do
+      result = Search.execute('"elephant"',
+        type_filter: 'topic',
+        include_blurbs: true
+      )
+
+      expect(result.posts).to contain_exactly(reply)
+      expect(result.blurb(reply)).to eq(expected_blurb)
+    end
+
+    it 'does not allow a post with repeated words to dominate the ranking' do
+      category = Fabricate(:category_with_definition, name: "winter is coming")
+
+      post = Fabricate(:post,
+        raw: "I think winter will end soon",
+        topic: Fabricate(:topic,
+          title: "dragon john snow winter",
+          category: category
+        )
+      )
+
+      post2 = Fabricate(:post,
+        raw: "I think #{'winter' * 20} will end soon",
+        topic: Fabricate(:topic, title: "dragon john snow summer", category: category)
+      )
+
+      result = Search.execute('winter')
+
+      expect(result.posts.pluck(:id)).to eq([
+        post.id, category.topic.first_post.id, post2.id
+      ])
+    end
+
+    it 'applies a small penalty to closed topic when ranking' do
+      post = Fabricate(:post,
+        raw: "My weekly update",
+        topic: Fabricate(:topic,
+          title: "A topic that will be closed",
+          closed: true
+        )
+      )
+
+      post2 = Fabricate(:post,
+        raw: "My weekly update",
+        topic: Fabricate(:topic,
+          title: "A topic that will be open"
+        )
+      )
+
+      result = Search.execute('weekly update')
+      expect(result.posts.pluck(:id)).to eq([post2.id, post.id])
+    end
+
+    it 'aggregates searches across topic correctly' do
```

Comment:
I would add a comment explaining what "correctly" means (eg. `shows the highest ranked post per topic`)

Reply:
Done :)